### PR TITLE
man: document how to use --network-interface= during boot

### PR DESCRIPTION
**Pull request by poettering**
Fixes: #18793

### CODE REVIEW

File: man/systemd-nspawn.xml [hunk 1]
```diff
@@ -801,46 +801,59 @@
...
-        <option>--private-network</option>. This option may be used
-        more than once to add multiple network interfaces to the
-        container.</para></listitem>
+        <listitem><para>Assign the specified network interface to the container. This will remove the
+        specified interface from the calling namespace and place it in the container. When the container
+        terminates, it is moved back to the calling namespace.  Note that
+        <option>--network-interface=</option> implies <option>--private-network</option>. This option may be
+        used more than once to add multiple network interfaces to the container.</para>
+
+        <para>Note that any network interface specified this way must exist already at the time the container
```
Comment (keszybz) [inline man/systemd-nspawn.xml]: must already exist ?

File: man/systemd-nspawn.xml [hunk 2]
```diff
@@ -801,46 +801,59 @@
...
+        is invoked. If the container shall be invoked automatically at boot via a
```
Comment (keszybz) [inline man/systemd-nspawn.xml]: invoked → started (I don't think we ever talk about *invoking* containers)

File: man/systemd-nspawn.xml [hunk 3]
```diff
@@ -801,46 +801,59 @@
...
+        (e.g. <filename>/etc/systemd/system/systemd-nspawn@foobar.service.d/50-network.conf</filename>) with
+        contents like the following:</para>
+
+        <programlisting>[Unit]
+Wants=sys-subsystem-net-devices-ens1.device
+After=sys-subsystem-net-devices-ens1.device</programlisting>
+
+        <para>This will make sure that activation of the container service will be delayed until the
+        <literal>ens1</literal> network interface has shown up. This is required since hardware probing is
+        fully asynchronous, and network interfaces might be discovered and probed only later during the boot
```
Comment (keszybz) [inline man/systemd-nspawn.xml]: "disovered and probed" → "discovered" ?

File: man/systemd-nspawn.xml [hunk 4]
```diff
@@ -801,46 +801,59 @@
...
-        more than once to add multiple network interfaces to the
-        container.</para></listitem>
+        <listitem><para>Create a <literal>macvlan</literal> interface of the specified Ethernet network
+        interface and add it to the container. A <literal>macvlan</literal> interface is a virtual interface
+        that adds a second MAC address to an existing physical Ethernet link. The interface in the container
+        will be named after the interface on the host, prefixed with <literal>mv-</literal>. Note that
+        <option>--network-macvlan=</option> implies <option>--private-network</option>. This option may be
+        used more than once to add multiple network interfaces to the container.</para>
+
+        <para>As with <option>--network-interface=</option> the underlying Ethernet network interface must
```
Comment (keszybz) [inline man/systemd-nspawn.xml]: Comma after "As with --network-interface"

File: man/systemd-nspawn.xml [hunk 5]
```diff
@@ -801,46 +801,59 @@
...
+        exist already at the time the container is started, and thus similar unit file drop-ins as described
```
Comment (keszybz) [inline man/systemd-nspawn.xml]: must already exist

File: man/systemd-nspawn.xml [hunk 6]
```diff
@@ -801,46 +801,59 @@
...
+        interface and add it to the container. An <literal>ipvlan</literal> interface is a virtual interface,
+        similar to a <literal>macvlan</literal> interface, which uses the same MAC address as the underlying
+        interface.  The interface in the container will be named after the interface on the host, prefixed
+        with <literal>iv-</literal>.  Note that <option>--network-ipvlan=</option> implies
+        <option>--private-network</option>. This option may be used more than once to add multiple network
+        interfaces to the container.</para>
+
+        <para>As with <option>--network-interface=</option> the underlying Ethernet network interface must
+        exist already at the time the container is started, and thus similar unit file drop-ins as described
+        above might be useful.</para></listitem>
```
Comment (keszybz) [inline man/systemd-nspawn.xml:856]: Ditto ditto

File: man/systemd-nspawn.xml [hunk 7]
```diff
@@ -907,7 +920,11 @@
         this option is used, the host side of the Ethernet link will use the <literal>vb-</literal> prefix
         instead of <literal>ve-</literal>. Regardless of the used naming prefix the same network interface
         name length limits imposed by Linux apply, along with the complications this creates (for details see
-        above).</para></listitem>
+        above).</para>
+
+        <para>As with <option>--network-interface=</option> the underlying Bridge network interface must
+        exist already at the time the container is started, and thus similar unit file drop-ins as described
+        above might be useful.</para></listitem>
```
Comment (keszybz) [inline man/systemd-nspawn.xml:927]: Ditto, Ditto

**Review by keszybz — CHANGES_REQUESTED** [review]
Looks nice.

**Comment by poettering** [comment]
Force pushed new version, only changes are the suggested ones. Upgrading green label.

**Pull request merged**